apm-agent-core/src/main/java/co/elastic/apm/agent/impl/transaction/TraceContext.java

```diff
@@ -95,6 +95,8 @@ public boolean asChildOf(TraceContext child, Object ignore) {
     private final Id transactionId = Id.new64BitId();
     private final StringBuilder outgoingHeader = new StringBuilder(TRACE_PARENT_LENGTH);
     private byte flags;
+    private boolean discard;
```

Comment:
also copy on `asChildOf`?

Reply:
No. Only `trace_methods` spans should be created as discardable. Their children shouldn't, unless the are such as well.